packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2210,7 +2213,7 @@ abstract class BuildContext {
   /// [InheritedWidget] subclasses that supports partial updates, like
   /// [InheritedModel]. It specifies what "aspect" of the inherited
   /// widget this context depends on.
-  T dependOnInheritedWidgetOfExactType<T extends InheritedWidget>({ Object aspect });
+  T? dependOnInheritedWidgetOfExactType<T extends InheritedWidget>({ Object aspect });
```

Comment:
`aspect` has to be nullable, no? I'm surprised this didn't either require "required" or "?".

Reply:
I was also surprised with that some times ago. IIRC this is allowed because the method is abstract? But that looks actually quite strange. /cc @leafpetersen

Reply:
Yes, see the 9th bullet in this section: https://github.com/dart-lang/language/blob/master/accepted/future-releases/nnbd/feature-specification.md#errors-and-warnings , landed [here](https://github.com/dart-lang/language/pull/733).

Reply:
Weird, what's the reasoning?

Reply:
@a14n well in any case aspect here should be nullable

Reply:
Done